### PR TITLE
Fixed more issues with the delete_group and paramater mismatch

### PR DESCRIPTION
**Pull request by toninog**
Fixed more issues with the delete_group and paramater mismatch

### CODE REVIEW

File: cloud/amazon/iam.py [hunk 1]
```diff
@@ -133,7 +133,7 @@
     iam_type: user
     name: jdavila
     state: update
-    groups: "{{ item.created_group.group_name }}"
+    group: "{{ item.created_group.group_name }}"
```
Comment (bcoca) [inline cloud/amazon/iam.py]: the correct one here is groups, not group

**Comment by bcoca** [comment]
cc @defionscode

**Comment by juliedavila** [comment]
@toninog is this just a style fix pr? or is there an issue with the module itself?

File: cloud/amazon/iam.py (hunk 1 above)
Comment (toninog) [inline cloud/amazon/iam.py]: The original commit I had had the groups and not group... it's weird it got reverted...

The other changes are to fix a bug that caused the delete of a group to work.  Below are the fixes for that

**Comment by toninog** [comment]
It was a module fix itself.  The delete_group function fails to delete unless my submitted fix is there.

**Comment by juliedavila** [comment]
@toninog can you fix/squash your commit so it's `groups` and not `group`

**Comment by toninog** [comment]
Sure.  Shall I do a new commit or try and merge the 3 commits ?

If merge : how do I do that ?

**Comment by juliedavila** [comment]
Ah, no. Make a commit to the same branch (no merging).

Then do `git rebase -i -p HEAD~2`

Then you should get a screen with two lines at the top with your commit messages.

The second line, on the far left should be changed to say `fixup` or `squash`.

The do a `git push -f`

**Comment by toninog** [comment]
ok - if you delete the PR I will resubmit a new PR with all the changes in

**Comment by toninog** [comment]
when running : git rebase -i -p HEAD-2 I see 

pick fccc925 fixes to code to enable updates of user to groups and delete groups.  Fixed example yaml to use groups
pick 9f3658d Fixed bug in example where updating a user to a group Fixed bug in create_group and update_group whereby paramaters were not being set correctly.
pick 68cfbca Fixed more issues with the delete_group and paramater mismatch
pick e49a90a Merge branch 'devel' of https://github.com/toninog/ansible-modules-core into devel

# Rebase ccfc9d2..e49a90a onto ccfc9d2

at the top

**Comment by juliedavila** [comment]
So normally you'd just be able to change all but the top most `pick` to `fixup` but in this case there is a merge commit which we can't accept AFAIK. Kind of a pain to do but you'll need to revert to remove the merge commit.

The simplest way would be to copy your patched up `iam.py` to say your home directory, rebranch (maybe even clone if you really want to start clean) off from devel and move over the iam.py again

**Comment by toninog** [comment]
ok - well then we can close this PR and I'll resubmit from a clean fork.

**Comment by juliedavila** [comment]
Ok, go ahead and close this then mention me in your new PR (sorry for the drag)

**Comment by toninog** [comment]
hey no worries.  I need a working iam module :)

**Pull request closed without merging**